src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,239 @@
+
+        const accessorModifiers = getAccessorModifiers(declaration, isStatic);
+        const fieldModifiers = getFieldModifiers(isStatic);
+
+        const getAccessor = generateGetAccessor(fieldName, accessorName, type, accessorModifiers, isStatic, container);
+        const setAccessor = generateSetAccessor(fieldName, accessorName, type, accessorModifiers, isStatic, container);
+
+        updateFieldDeclaration(changeTracker, file, declaration, fieldName, fieldModifiers, container);
+
+        insertAccessor(changeTracker, file, getAccessor, declaration, container);
+        insertAccessor(changeTracker, file, setAccessor, declaration, container);
+
+        const edits = changeTracker.getChanges();
+        const renameFilename = file.fileName;
+        const renameLocationOffset = isIdentifier(fieldName) ? 0 : -1;
+        const renameLocation = renameLocationOffset + getRenameLocation(edits, renameFilename, fieldName.text, /*isDeclaredBeforeUse*/ false);
+        return { renameFilename, renameLocation, edits };
+    }
+
+    function getAccessorModifiers(declaration: AccepedDeclaration, isStatic: boolean): NodeArray<Modifier> {
+        if (!declaration.modifiers || getModifierFlags(declaration) & ModifierFlags.Private) {
+            return createNodeArray(
+                append<Modifier>([createToken(SyntaxKind.PublicKeyword)],
+                    isStatic ? createToken(SyntaxKind.StaticKeyword) : undefined));
+        }
+        return declaration.modifiers;
+    }
+
+    function getFieldModifiers(isStatic: boolean): NodeArray<Modifier> {
+        return createNodeArray(
+            append<Modifier>([createToken(SyntaxKind.PrivateKeyword)],
+                isStatic ? createToken(SyntaxKind.StaticKeyword) : undefined));
+    }
+
+    function isConvertableName (name: DeclarationName): name is AccepedNameTypes {
+        return isIdentifier(name) || isStringLiteral(name);
+    }
+
+    function getPropertyDeclarationInfo(propertyDeclaration: PropertyDeclaration): DeclarationInfo | undefined {
+        if (!isClassLike(propertyDeclaration.parent) || !propertyDeclaration.parent.members) return undefined;
+
+        return {
+            isStatic: hasStaticModifier(propertyDeclaration),
+            type: propertyDeclaration.type,
+            container: propertyDeclaration.parent
+        };
+    }
+
+    function getParameterPropertyDeclarationInfo(parameterDeclaration: ParameterDeclaration): DeclarationInfo | undefined {
+        if (!isClassLike(parameterDeclaration.parent.parent) || !parameterDeclaration.parent.parent.members) return undefined;
+
+        return {
+            isStatic: false,
+            type: parameterDeclaration.type,
+            container: parameterDeclaration.parent.parent
+        };
+    }
+
+    function getPropertyAssignmentDeclarationInfo(propertyAssignment: PropertyAssignment): DeclarationInfo | undefined {
+        return {
+            isStatic: false,
+            type: undefined,

```

Comment:
what should i do with this type declaration,
should i get the type with typeChecker?

Reply:
no, you are fine. the type will be inferred from the return type of the getter.